graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/scriptingapi/mapping/SearchJobToTabularResponseMapper.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.plugins.views.search.rest.scriptingapi.mapping;
+
+import com.google.common.collect.ImmutableList;
+import org.apache.commons.collections.CollectionUtils;
+import org.graylog.plugins.views.search.QueryResult;
+import org.graylog.plugins.views.search.SearchJob;
+import org.graylog.plugins.views.search.SearchType;
+import org.graylog.plugins.views.search.errors.SearchError;
+import org.graylog.plugins.views.search.rest.SearchJobDTO;
+import org.graylog.plugins.views.search.rest.scriptingapi.request.Metric;
+import org.graylog.plugins.views.search.rest.scriptingapi.request.SearchRequestSpec;
+import org.graylog.plugins.views.search.rest.scriptingapi.response.Metadata;
+import org.graylog.plugins.views.search.rest.scriptingapi.response.ResponseSchemaEntry;
+import org.graylog.plugins.views.search.rest.scriptingapi.response.TabularResponse;
+import org.graylog.plugins.views.search.searchtypes.pivot.PivotResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.NotFoundException;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class SearchJobToTabularResponseMapper {
```

Comment:
1. It does not need to happen in this PR, but we may want to unit test this class.
2. I am not 100% sure if we want to to use "javax.ws.rs.*" exceptions in this mapper, or should it rather be unaware/independent of this package, throwing more general exceptions, that will be mapped to proper exceptions by the resource class - just to consider.

Reply:
yeah, custom exceptions and their mappers would be the cleanest solution. Let me do that here when we are already refactoring error handling.

Reply:
added custom exception that's re-thrown in the resource. Primitive but cleaner than before.